pkg/handlers/webhookreceiver.go

```diff
@@ -104,8 +106,11 @@ func (h *WebhookReceiverHandler) ServeHTTP(w http.ResponseWriter, r *http.Reques
 	if err != nil {
 		log.Errorf("Failed to write to response: %s\n", err)
 		http.Error(w, "Failed to write to response", http.StatusInternalServerError)
+		metrics.SetRequestMetricFailure(WebhookReceiverPath)
 		return
 	}
+
+	metrics.ResetMetric("metricRequestFailure")
```

Comment:
Once there's a string being used to in two separate functions which impacts logic flow, it might be a sign to turn it into a constant that both functions can refer to. Maybe turn this string and `"metricResponseFailure"` to constants which this func and the `ResetMetric` one can use?

Reply:
Another solution is we just declare the variable `metricRequestFailure` and `metricResponseFailure` as `MetricRequestFailure` and `MetricResponseFailure`.

Then the reset func will be
```
func ResetMetric(m *prometheus.GaugeVec) {
	m.Reset()
}
```

And we can call it from the webhookreceiver.go with 
```
	metrics.ResetMetric(metrics.MetricRequestFailure)

```
WDYT?

Reply:
Ah that also looks good